pynest/nest/tests/test_csa.py

```diff
@@ -48,129 +48,125 @@
-                nest.GetConnections([targets[i]]), 'target')
+            conns = nest.GetStatus(nest.GetConnections([targets[i]]))
             self.assertEqual(len(conns), 0)
 
-    def test_CSA_OneToOne_params(self):
-        """One-to-one connectivity"""
+    def test_CSA_OneToOne_synmodel(self):
+        """One-to-one connectivity with synmodel"""
 
         nest.ResetKernel()
 
-        n = 4  # number of neurons
+        n_neurons = 4
+        synmodel = "stdp_synapse"
 
-        pop0 = nest.LayoutNetwork("iaf_neuron", [n])
-        pop1 = nest.LayoutNetwork("iaf_neuron", [n])
+        sources = nest.Create("iaf_psc_alpha", n_neurons)
+        targets = nest.Create("iaf_psc_alpha", n_neurons)
 
-        cs = csa.cset(csa.oneToOne, 10000.0, 1.0)
+        cs = csa.cset(csa.oneToOne)
 
-        nest.CGConnect(pop0, pop1, cs, {"weight": 0, "delay": 1})
+        nest.CGConnect(sources, targets, cs, model=synmodel)
 
-        sources = nest.GetLeaves(pop0)[0]
-        targets = nest.GetLeaves(pop1)[0]
-        for i in range(n):
-            conns = nest.GetStatus(
-                nest.GetConnections([sources[i]]), 'target')
+        for i in range(n_neurons):
+            conns = nest.GetStatus(nest.GetConnections([sources[i]]))
             self.assertEqual(len(conns), 1)
-            self.assertEqual(conns[0], targets[i])
+            self.assertEqual(conns[0]["target"], targets[i])
+            self.assertEqual(conns[0]["synapse_model"], synmodel)
 
-            conns = nest.GetStatus(
-                nest.GetConnections([targets[i]]), 'target')
+            conns = nest.GetStatus(nest.GetConnections([targets[i]]))
             self.assertEqual(len(conns), 0)
 
-    @unittest.skipIf(not HAVE_NUMPY, 'NumPy package is not available')
-    def test_CSA_cgnext(self):
-        """cgnext"""
+    def test_CSA_error_handling(self):
+        """Error handling of CGConnect"""
 
         nest.ResetKernel()
 
-        w = 10000.0
-        d = 1.0
-        cs = csa.cset(csa.oneToOne, w, d)
-
-        nest.sli_push(cs)
-        nest.sli_run('dup')
-        nest.sli_push(numpy.array([0, 1, 2, 3]))
-        nest.sli_push(numpy.array([0, 1, 2, 3]))
-        nest.sli_run('cgsetmask')
-        nest.sli_run('dup')
-        nest.sli_run('cgstart')
-        for i in range(4):
-            nest.sli_run('dup')
-            nest.sli_run('cgnext')
-            self.assertEqual(nest.sli_pop(), True)
-            self.assertEqual(nest.sli_pop(), d)
-            self.assertEqual(nest.sli_pop(), w)
-            self.assertEqual(nest.sli_pop(), i)
-            self.assertEqual(nest.sli_pop(), i)
-        nest.sli_run('cgnext')
-        self.assertEqual(nest.sli_pop(), False)
+        cs = csa.cset(csa.oneToOne)
+        nonnodes = [1, 2, 3]
+
+        self.assertRaisesRegex(nest.NESTError, "UnknownNode",
+                               nest.CGConnect, nonnodes, nonnodes, cs)
+
+        n_neurons = 4
+
+        sources = nest.Create("iaf_psc_alpha", n_neurons)
```

Comment:
You have two tests in a single unit-test. Depending on the unittest framework this might cause the second tests to be skipped if the first one fails.

Reply:
Right. Thanks. I've split them in 15411851c.